04/app.js

```diff
@@ -12,11 +13,38 @@ class App extends React.Component {
     render() {
         return (
             <section>
-                <File />
-                <List />
+                <File onChange={this.addFileDataToState}/>
+                <List data={this.state.filesList}/>
             </section>
         )
     }
+
+    addFileDataToState = (data) => {
+        Object.values(data).map( file => {
+            if( file && file.type.includes('text') ) {
+
+                const reader = new FileReader();
+
+                reader.onload = (readerEvent) => {
+                    const fileData = {
+                        name: file.name,
+                        size: file.size,
+                        content: readerEvent.target.result,
+                        id: uuid()
```

Comment:
ooo widzisz! Tutaj jest ok, bo dodajesz dany element, przypisujesz mu `id` i przy renderowaniu to `id` się już nie zmienia.